### PR TITLE
Symlink input using element_identifier

### PR DESCRIPTION
**Pull request by mvdbeek**
We symlink input files into the current working directory using their
element_identifier (which falls back to the name for non-collection
elements) for most tools that have straightforward single inputs. This is
beneficial for tools the produce reports like the MarkDuplicates and the
*Metrics tools.

### CODE REVIEW

File: tools/picard/picard_DownsampleSam.xml [hunk 1]
```diff
@@ -51,19 +52,19 @@ Randomly down-sample a SAM or BAM file to retain a random subset of the reads. M
 
 @description@
 
-  INPUT=File
-  I=File              The input SAM or BAM file to downsample.  Required. 
+  INPUT=$inputFile.element_identifierFile
```
Comment (nsoranzo) [inline tools/picard/picard_DownsampleSam.xml]: Here too.

File: tools/picard/picard_AddOrReplaceReadGroups.xml [hunk 2]
```diff
@@ -65,43 +66,43 @@ Add or Replace Read Groups in an input BAM or SAM file.
 
 @description@
 
-  INPUT=File
-  I=File                  Input file (bam or sam).  Required. 
+  INPUT=$inputFile.element_identifierFile
```
Comment (mvdbeek) [inline tools/picard/picard_AddOrReplaceReadGroups.xml]: Oops, yeah, that sholdn't be there.

**Comment by bgruening** [comment]
:+1:

**Comment by mvdbeek** [comment]
Thanks @bgruening ... this should be quite helpful for https://github.com/EnginesOn/galaxy_tools/tree/master/tools/multiqc / https://toolshed.g2.bx.psu.edu/repository?repository_id=9456ff7806d35888&changeset_revision=66f1187ed363.

**Comment by bgruening** [comment]
Oh yes it is. @yvanlebras is developing it with his colleagues. I think they have also a Galaxy-Tour for it. Not sure about the plans to move it to IUC ... but we discussed this once.

File: tools/picard/picard_macros.xml [hunk 3]
```diff
@@ -16,6 +16,10 @@
         </requirements>
     </xml>
 
+    <token name="@symlink_element_identifier@"><![CDATA[
+    ln -f -s '$inputFile' '$inputFile.element_identifier' &&
+    ]]></token>
```
Comment (jmchilton) [inline tools/picard/picard_macros.xml]: This must be escaped in my opinion https://github.com/galaxyproject/tools-iuc/pull/1019.

**Pull request merged**